### PR TITLE
`SurfaceCurrentRegularization` regularization options

### PR DESCRIPTION
**Pull request by ddudt**
Adds regularization options to the `SurfaceCurrentRegularization` objective. This is similar to the `regularization_type` argument of the function `solve_regularized_surface_current`.

### CODE REVIEW

**Comment by ddudt** [comment]
@ddudt remember to update units and normalization

File: desc/objectives/_coils.py [hunk 1]
```diff
@@ -2457,5 +2467,10 @@ def compute(self, surface_params=None, constants=None):
             profiles={},
         )
 
-        K_mag = safenorm(surface_data["K"], axis=-1)
-        return K_mag * jnp.sqrt(surface_data["|e_theta x e_zeta|"])
+        if self._regularization == "K":
+            K = safenorm(surface_data["K"], axis=-1)
+        elif self._regularization == "Phi":
+            K = jnp.abs(surface_data["Phi"])
```
[the remark this answers is not in the document]
Comment (dpanici) [reply desc/objectives/_coils.py:2473]: I guess it doesn't matter and this way the printout is more legible

**Review by dpanici — APPROVED** [review]

File: desc/objectives/_coils.py [hunk 2]
```diff
@@ -2292,6 +2292,8 @@ class SurfaceCurrentRegularization(_Objective):
     surface_current_field : CurrentPotentialField
         Surface current which is producing the magnetic field, the parameters
         of this will be optimized to minimize the objective.
+    regularization : str, optional
+        Regularization method. One of {'K', 'Phi', 'sqrt(Phi)'}. Default = 'K'.
```
Comment (f0uriest) [inline desc/objectives/_coils.py:2296]: should also modify the formula above on 2267

File: desc/objectives/_coils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddudt) [reply desc/objectives/_coils.py:2491]: It's not necessary and would probably be more general without the absolute value. But since it doesn't really matter and the other two regularization options are always positive, I prefer to have it.

**Review by YigitElma — DISMISSED** [review]

File: desc/objectives/_coils.py [hunk 3]
```diff
@@ -2262,6 +2262,8 @@ def compute(self, params, constants=None):
 class SurfaceCurrentRegularization(_Objective):
     """Target the surface current magnitude.
 
+    If ``regularization="k"``:
```
Comment (dpanici) [inline desc/objectives/_coils.py:2265]:
```suggestion
    If ``regularization="K"``:
```

File: desc/objectives/_coils.py [hunk 4]
```diff
@@ -2292,6 +2306,8 @@ class SurfaceCurrentRegularization(_Objective):
     surface_current_field : CurrentPotentialField
         Surface current which is producing the magnetic field, the parameters
         of this will be optimized to minimize the objective.
+    regularization : str, optional
+        Regularization method. One of {'K', 'Phi', 'sqrt(Phi)'}. Default = 'K'.
```
Comment (dpanici) [inline desc/objectives/_coils.py:2310]:
```suggestion
        Regularization method. One of {'"K"', '"Phi"', '"sqrt(Phi)"'}. Default = '"K"'.
```

**Review by dpanici — APPROVED** [review]

**Review by dpanici — CHANGES_REQUESTED** [review]
clarify docstring slightly, I left suggested commit

**Review by dpanici — APPROVED** [review]

**Review by YigitElma — APPROVED** [review]
